LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection.js

```diff
@@ -522,24 +600,69 @@ ReflectionTests.typeMap = {
-                        null, null, 1.5, 5, 100, 0.5, null, null,
-                        null, null, null, null, null,
-                        2, 3],
-        // I checked that ES ToString is well-defined for all of these (I
-        // think).  Yes, String(-0) == "0".
+            {valueOf:function() {return 3;}, toString: null}],
+        "domExpected": function (val) {
+            var parsed = ReflectionTests.parseFloat(String(val));
+            if (parsed === false) {
+                return null;
+            }
+            return parsed;
+        },
+        "idlTests":       [ -10000000000,   -1,  -0,   0,   1,   10000000000],
+        "idlDomExpected": function (val) {
+            return Number(val).toFixed();
+        },
+        "idlIdlExpected": function (val) {
+            // This is a bit heavy-weight but hopefully will give values
+            // that compare "better" (without introducing some tolerance)
+            // when the test cases are expanded with more values.
+            return ReflectionTests.parseFloat(Number(val).toFixed());
+        }
+    },
+    /**
+     * Reflected IDL attribute of type double, limited to only positive values,
+     * are similar to the previous case with the following exceptions:
+     *
+     *  - on getting, if the parsed value is not greater than 0, then return
+     *    the default value
+     *  - on setting, if the value is not greater than 0, then return (leaving)
+     *    the attribute to its previous value.
+     */
+    "limited double": {
+        "jsType": "number",
+        "defaultVal": 0.0,
+        "domTests": [minInt - 1, minInt, -36, -1, 0, 1, maxInt,
+            maxInt + 1, maxUnsigned, maxUnsigned + 1, "",
+            "\u00097", "\u000B7", "\u000C7", "\u00207", "\u00A07", "\uFEFF7",
+            "\t\u000B7", "\n\u000B7","\f\u000B7", "\r\u000B7", "\x20\u000B7", "7\u000B",
+            "\u000A7", "\u000D7", "\u20287", "\u20297", "\u16807", "\u180E7",
+            "\u20007", "\u20017", "\u20027", "\u20037", "\u20047", "\u20057",
+            "\u20067", "\u20077", "\u20087", "\u20097", "\u200A7", "\u202F7",
+            "\u30007",
+            " " + binaryString + " foo ", undefined, 1.5, "5%", "+100", ".5", true, false,
+            {"test": 6}, NaN, +Infinity, -Infinity, "\0",
+            {toString:function() {return 2;}, valueOf: null},
+            {valueOf:function() {return 3;}, toString: null}],
+        "domExpected": function (val) {
+            var parsed = ReflectionTests.parseFloat(String(val));
+            if (parsed === false || parsed <= 0) {
+                return null;
+            }
+            return parsed;
+        },
         "idlTests":       [ -10000000000,   -1,  -0,   0,   1,   10000000000],
-        "idlDomExpected": ["-10000000000", "-1", "0", "0", "1", "10000000000"],
-        "idlIdlExpected": [ -10000000000,   -1,  -0,   0,   1,   10000000000]
+        "idlDomExpected": function (val) {
+            return Number(val).toFixed();
```

Comment:
Fwiw, I updated the WPT pull request, as this was actually wrong (won't make a difference for the 5 values being tested though).